app/controllers/local_authorities_controller.rb

```diff
@@ -2,4 +2,27 @@ class LocalAuthoritiesController < ApplicationController
   def index
     @authorities = LocalAuthority.all.order(name: :asc)
   end
+
+  def edit
+    @authority = LocalAuthority.find_by(slug: params[:slug])
+  end
+
+  def update
+    authority = LocalAuthority.find_by(slug: params[:slug])
+    authority.homepage_url = params[:local_authority][:homepage_url]
+    validate_and_save(authority)
+  end
+
+private
+
+  def validate_and_save(authority)
```

Comment:
What's been said already about `valid?` and `save!` and `redirect_to` in the `LinksController` applies here too.